lang/syn/src/idl/external.rs

```diff
@@ -17,8 +17,8 @@ pub fn get_external_type(name: &str, path: impl AsRef<Path>) -> Result<Option<sy
         .ok_or_else(|| anyhow!("`{name}` not found in use statements"))?;
 
     // Get crate name and version from lock file
-    let lib_path = find_path("lib.rs", path)?;
-    let lock_path = find_path("Cargo.lock", lib_path)?;
+    let program_path = std::env::var("ANCHOR_IDL_BUILD_PROGRAM_PATH").expect("Failed to get program path");
```

Comment:
Question: would this work with `cargo`'s `CARGO_MANIFEST_DIR` env varible? My intuition says no, but if that also works, it would allow us to get rid of the `ANCHOR_IDL_BUILD_PROGRAM_PATH` env varible.

Reply:
Yes, it doesn't work, since when a dependency is getting built CARGO_MANIFEST_DIR is the `Cargo.toml` for the dependency in the cargo local registry.